### PR TITLE
Add media-feature-name-case

### PR DESCRIPTION
**Pull request by evilebottnawi**
Let's rock with media queries rules :tada: 
Ref: https://github.com/stylelint/stylelint/issues/1031
/cc @davidtheclark @jeddy3 @ntwb

### CODE REVIEW

File: src/rules/media-feature-name-case/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# media-feature-name-case
+
+Specify lowercase or uppercase for media feature name.
+
+```css
```
Comment (jeddy3) [inline src/rules/media-feature-name-case/README.md]: Everything flush to left.

Comment (evilebottnawi) [inline src/rules/media-feature-name-case/README.md]: @jeddy3 oh sorry :smile: yep

File: src/utils/isStandardSyntaxMediaFeature.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/**
```
Comment (jeddy3) [inline src/utils/isStandardSyntaxMediaFeature.js]: The terminology doesn't seem quite right here. What actually being tested is a validity of a [_media feature **name**_](https://www.w3.org/TR/mediaqueries-4/#mq-features), not the whole *media feature*.

So, I think we either rename this to `isStandardSyntaxMediaFeatureName` or we adapt this util to check the validity of the whole media feature e.g. `(min-width: 10px)`. I prefer the latter as this rule targets *media features* and so we should exclude anything that isn't a media feature *with standard syntax*.

Is the terminology right in the parser itself?

File: src/rules/media-feature-name-case/index.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        const mediaFeatureName = mediaFeatureNode.value
+
+        const expectedFeatureName = expectation === "lower"
+          ? mediaFeatureName.toLowerCase()
+          : mediaFeatureName.toUpperCase()
+
+        if (mediaFeatureName === expectedFeatureName) { return }
+
+        report({
+          index: atRule.name.length + atRule.raws.afterName.length + mediaFeatureNode.sourceIndex + 1,
```
Comment (jeddy3) [inline src/rules/media-feature-name-case/index.js]: Can we not use the `atRuleParamIndex` util here?

**Comment by jeddy3** [comment]
@evilebottnawi It's looking really good for the get-go. I've some queries about terminology and ignoring non-standard syntax.

File: src/utils/isStandardSyntaxMediaFeature.js (hunk 2 above)
Comment (evilebottnawi) [inline src/utils/isStandardSyntaxMediaFeature.js]: @jeddy3 yes, you think create `isStandardSyntaxMediaFeature` (my rename to `isStandardSyntaxMediaFeatureName`) in this PR? or other?

File: src/rules/media-feature-name-case/index.js (hunk 3 above)
Comment (evilebottnawi) [inline src/rules/media-feature-name-case/index.js]: @jeddy3 thank, try use `atRuleParamIndex`

File: src/utils/isStandardSyntaxMediaFeature.js (hunk 2 above)
Comment (jeddy3) [inline src/utils/isStandardSyntaxMediaFeature.js]: >  in this PR? 

Sure.

File: src/rules/media-feature-name-case/README.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
+# media-feature-name-case
+
+Specify lowercase or uppercase for media feature name.
```
Comment (ntwb) [inline src/rules/media-feature-name-case/README.md]: This should be plural: `Specify lowercase or uppercase for media feature names.`, i.e. `names.`, not `name.`

File: src/rules/media-feature-name-case/README.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+```css
+    @media (min-width: 700px) { }
+/**         ↑
+ *          These media feature name */
```
Comment (ntwb) [inline src/rules/media-feature-name-case/README.md]: Plural again here please, `These media feature names`

**Comment by evilebottnawi** [comment]
@jeddy3 all works fine, besides create `isStandardSyntaxMediaFeature` is not yet possible. Nodes inside `media-feature` not have `parent` and i can't get `media-feature` for check. What do you think?
/cc @davidtheclark need review and look what i say above

**Comment by jeddy3** [comment]
LGTM.

Lets write `isStandardSyntaxMediaFeature` in https://github.com/stylelint/stylelint/pull/1652

File: src/rules/media-feature-name-case/__tests__/index.js [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+
+testRule(rule, {
+  ruleName,
+  config: ["upper"],
+  syntax: "less",
+
+  accept: [ {
+    code: "@media (MIN-WIDTH: @tablet) { }",
+  }, {
+    code: "@media (MIN-WIDTH: (@value + 10px)) { }",
```
Comment (davidtheclark) [inline src/rules/media-feature-name-case/__tests__/index.js]: Why are there these tests that are equivalent to other tests but with extra parentheses? What do they add?

File: src/utils/isStandardSyntaxMediaFeatureName.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+ * Check whether a media feature is standard
+ *
+ * @param {string} media feature value
+ * @return {boolean} If `true`, the declaration is standard
+ */
+export default function (mediaFeatureValue) {
+
+  if (mediaFeatureValue[0] === "-" && mediaFeatureValue[1] === "-") { return false }
+
+  if (/#{.+?}|\$.+?/.test(mediaFeatureValue)) { return false }
```
Comment (davidtheclark) [inline src/utils/isStandardSyntaxMediaFeatureName.js]: Please add comments explaining why these two lines return false.

**Comment by davidtheclark** [comment]
Just a couple of comments, then it's fine to rebase and merge.

File: src/utils/isStandardSyntaxMediaFeatureName.js (hunk 7 above)
Comment (jeddy3) [inline src/utils/isStandardSyntaxMediaFeatureName.js]: Can we also rename the variables and correct the docs please:

```js
/**
 * Check whether a media feature name is standard
 *
 * @param {string} media feature name
 * @return {boolean} If `true`, the media feature name is standard
 */
export default function (mediaFeatureName) {

  // What this does
  if (mediaFeatureName[0] === "-" && mediaFeatureName[1] === "-") { return false }

  // What this does
  if (/#{.+?}|\$.+?/.test(mediaFeatureName)) { return false }

  return true
}
```

File: src/rules/media-feature-name-case/__tests__/index.js (hunk 6 above)
Comment (evilebottnawi) [inline src/rules/media-feature-name-case/__tests__/index.js]: @davidtheclark in scss if you use `(` and `)`, scss interpreted this as mathematical expression `(@value + 10px)` (if `@value: 10px`) = `20px`, without `(` and `)` scss interpreted as string `10px10px`

**Pull request merged**